Update documentation for v3 new features (#296)
aerospike/aerospike-backup-service: README.md

  The specific backup data location is now specified using a new required field: **`backup-data-path`**.
  This change allows you to reuse the same storage for different restore requests.

### Backup Retention Policy

This release introduces a new, configurable **RetentionPolicy** for managing backup storage more effectively.

> The feature allows users to specify retention rules for both full and incremental backups, replacing the previous rigid options (`KeepAll`, `RemoveAll`, `RemoveIncremental`).

Reviewer: ```suggestion
The feature allows users to specify retention rules for both full and incremental backups, replacing the previous options `KeepAll`, `RemoveAll`, and `RemoveIncremental`.
```

> Retention policy is (optional) part of Backup Policy. It consists of 2 integer fields.

Reviewer: ```suggestion
Retention policy is an optional part of a backup policy. It consists of two integer fields:
```

> * `full`: The total number of full backups to retain. If absent, all full backups are kept. Minimum is 1, it means that each new full backup will delete the previous one.

Reviewer: ```suggestion
* `full`: The total number of full backups to retain. If not specified, all full backups are kept. The minimum is 1, meaning each new full backup deletes the previous one.
```

> * `incremental`: The number of full backups to retain their incremental backups. Cannot exceed `full`. Minimum is 0, it means that all old incremental backups will be deleted after each full. 

Reviewer: I'm not clear exactly on how this parameter works, can you give more examples in a comment of what would happen with a value of 1, or what happens when it's the same as `full`? What happens if the user enters a value equal to the value of `full`?

Author: nil means no deletion.
full = nil, incr = nil => keep all
full = nil, incr = 0 => keep all full, delete all incremental after next full backup. So, incremental backups will actually happen, but they will live only up to next full backup.
full = 0 => forbidden
full = 1 => only store one full backup
full = N, incr = M => store N full backups, for last M of them store also incremental backups. N >= M.
full = N, incr = N => store N full backups, all with incrementals.

Does that make sense?

Reviewer: Got it, I think I would rewrite it as such, please check if this is right:

`* `incremental`: The number of most recent full backups that also retain intermediate backups made between them. Cannot exceed the value of `full`. If omitted, all incremental backups are kept. A value of 0 means that all previous existing incremental backups will be deleted after each full backup is made.`

Please also add one or two lines saying what happens with the forbidden cases of "full = 0" and "incremental > full". Error? Fall back to default of nil?

Author: We never used word "intermediate" before, let's stick with "incremental".
In forbidden cases it will fail validation on startup. Just like any other field, we have a lot of validation rules.

Reviewer: Yes, you're right of course on "incremental." I'll add something in the main docs about validation checks, no need to add it here. Copy in that wording and I'll re-review, thanks!

> If no RetentionPolicy is specified, the system will default to retaining all backups, mimicking the previous KeepAll behavior.

Reviewer: ```suggestion
If no retention policy is specified, the system defaults to retaining all full and incremental backups, the same as the `KeepAll` value in older versions.
```

After each successfull full backup, all existing backups are scanned to count full and incremental backups.

> Service then removes older full backups and their associated incrementals as needed to retain only

Reviewer: ```suggestion
ABS then removes older full backups and their associated incremental backups as needed to retain only
```

> the last `full` backups and incremental backups for the most recent `incemental` backups.

Reviewer: ```suggestion
the last `full` backups and incremental backups for the most recent `incremental` backups.
```

This release introduces a new, configurable **RetentionPolicy** for managing backup storage more effectively.
The feature allows users to specify retention rules for both full and incremental backups, replacing the previous options `KeepAll`, `RemoveAll`, and `RemoveIncremental`.
Retention policy is an optional part of a backup policy. It consists of two integer fields:

* `full`: The total number of full backups to retain. If not specified, all full backups are kept. The minimum is 1, meaning each new full backup deletes the previous one.

> * `incremental`: The number of full backups to retain their incremental backups. Cannot exceed `full`. Minimum is 0, it means that all old incremental backups will be deleted after each full. 

Reviewer: ```suggestion
* `incremental`: The number of most recent full backups that also retain incremental backups made between them. Cannot exceed the value of `full`. If omitted, all incremental backups are kept. A value of `0` means that all previous existing incremental backups will be deleted after each full backup is made.
```

    type: object
```

### Cancel Restore Job

New endpoint:  
[`POST {{baseUrl}}/v1/restore/cancel/:<jobId>`](https://aerospike.github.io/aerospike-backup-service/#/Restore/cancel)

Cancel the restore job identified by `<jobId>`. Data that has already been restored will remain intact.

### Cancel Backup Job

New endpoint:  
[`POST {{baseUrl}}/v1/backups/cancel/:<routineName>`](https://aerospike.github.io/aerospike-backup-service/#/Backup/cancel)

Cancel all currently running backups (both full and incremental) for the specified routine. Partially created backups will be deleted.

### Disable Routine

New endpoints:

- [`POST {{baseUrl}}/v1/routines/:<routineName>/disable/`](https://aerospike.github.io/aerospike-backup-service/#/Routine/disable)
- [`POST {{baseUrl}}/v1/routines/:<routineName>/enable/`](https://aerospike.github.io/aerospike-backup-service/#/Routine/enable)

> Set the disabled flag for the given routine to `true` or `false` (default is `false`).

Reviewer: These github.io links don't go to the correct section on the live page. Can you update them, or will the openAPI spec get updated to reflect this?

Author: Yes, the API is not published yet.
